larpandoracontent/LArTwoDReco/LArClusterAssociation/ClusterAssociationAlgorithm.cc

```diff
@@ -194,69 +198,29 @@ void ClusterAssociationAlgorithm::UpdateForUnambiguousMerge(const Cluster *const
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-void ClusterAssociationAlgorithm::UpdateForAmbiguousMerge(const Cluster *const pClusterToEnlarge, const Cluster *const pClusterToDelete, const bool isForwardMerge,
-    ClusterAssociationMap &clusterAssociationMap) const
+void ClusterAssociationAlgorithm::UpdateForAmbiguousMerge(const Cluster *const pCluster, ClusterAssociationMap &clusterAssociationMap) const
 {
-    ClusterAssociationMap::iterator iterEnlarge = clusterAssociationMap.find(pClusterToEnlarge);
-    ClusterAssociationMap::iterator iterDelete = clusterAssociationMap.find(pClusterToDelete);
+    ClusterAssociationMap::iterator cIter = clusterAssociationMap.find(pCluster);
 
-    if ((clusterAssociationMap.end() == iterEnlarge) || (clusterAssociationMap.end() == iterDelete))
+    if (clusterAssociationMap.end() == cIter)
         throw StatusCodeException(STATUS_CODE_NOT_FOUND);
 
-    ClusterSet &clusterSetEnlarge(isForwardMerge ? iterEnlarge->second.m_forwardAssociations : iterEnlarge->second.m_backwardAssociations);
-    ClusterSet &clusterSetDelete(isForwardMerge ? iterDelete->second.m_backwardAssociations : iterDelete->second.m_forwardAssociations);
-
-    for (ClusterSet::iterator iter = clusterSetEnlarge.begin(); iter != clusterSetEnlarge.end();)
-    {
-        if ((*iter) != pClusterToDelete)
-        {
-            ClusterAssociationMap::iterator iterAssociation = clusterAssociationMap.find(*iter);
-
-            if (clusterAssociationMap.end() == iterAssociation)
-                throw StatusCodeException(STATUS_CODE_NOT_FOUND);
-
-            ClusterSet &associatedClusterSet(isForwardMerge ? iterAssociation->second.m_backwardAssociations : iterAssociation->second.m_forwardAssociations);
-
-            ClusterSet::iterator enlargeIter = associatedClusterSet.find(pClusterToEnlarge);
-
-            if (associatedClusterSet.end() == enlargeIter)
-                throw StatusCodeException(STATUS_CODE_NOT_FOUND);
-
-            associatedClusterSet.erase(enlargeIter);
-            clusterSetEnlarge.erase(iter++);
-        }
-        else
-        {
-            ++iter;
-        }
-    }
-
-    for (ClusterSet::iterator iter = clusterSetDelete.begin(); iter != clusterSetDelete.end();)
+    for (ClusterAssociationMap::iterator mIter = clusterAssociationMap.begin(), mIterEnd = clusterAssociationMap.end(); mIter != mIterEnd; ++mIter)
```

Comment:
We could sort the clusters before looping over them here. It shouldn't make any difference, but means this should always happen in the same order (a nice thing, especially if anything unexpected happens when e.g. if this code was modified in future).